library/core/src/alloc/layout.rs

```diff
@@ -130,6 +130,8 @@ impl Layout {
     }
 
     /// The minimum byte alignment for a memory block of this layout.
+    ///
+    /// The returned alignment is guaranteed to be a non-zero power of two.
```

Comment:
I think this is sufficient without "non-zero". Zero is not a power of two.

https://doc.rust-lang.org/1.72.1/std/alloc/struct.Layout.html#method.from_size_align lists _"must not be zero"_ and _"must be a power of two"_ separately I guess because it's super important that it can't be misunderstood by someone who might be shaky on what power-of-two means. But here it's the standard library providing a guarantee to the caller, instead of the caller providing a guarantee to the standard library.

```suggestion
    /// The returned alignment is guaranteed to be a power of two.
```